libnetwork/pasta/pasta_linux.go

```diff
@@ -110,13 +110,25 @@ func Setup(opts *SetupOptions) (*SetupResult, error) {
 			return err
 		}
 		for _, addr := range addrs {
-			// make sure to skip localhost and other special addresses
-			if ipnet, ok := addr.(*net.IPNet); ok && ipnet.IP.IsGlobalUnicast() {
-				result.IPAddresses = append(result.IPAddresses, ipnet.IP)
-				if !ipv4 && util.IsIPv4(ipnet.IP) {
+			// make sure to skip loopback and multicast addresses
+			if ipnet, ok := addr.(*net.IPNet); ok && !ipnet.IP.IsLoopback() && !ipnet.IP.IsMulticast() {
```

Comment:
Are unspecified (0.0.0.0, IP.IsUnspecified()) addresses already filtered somewhere else?

Reply:
It is not filtered you are right. But I don't we have to either. We dump the addresses after the pasta setup in a new namespace so I doubt we ever get a unspecified address. I didn't test but I assume the kernel would not allow anyone to assign 0.0.0.0 to an interface as such adding the extra check here didn't seem useful.
But I guess it wouldn't hurt either.

Reply:
Ah, sorry, never mind, I totally misinterpreted this, it comes from `net.InterfaceAddrs()`!

Reply:
No no, there's no need to check.